DiscordRPC: Fix bug where discord would crash the game if you close it

DarthMDev opened the pull request:
Fixes issue #174

This pull request includes several changes to improve the Discord Rich Presence functionality in the `toontown/discord/DiscordRPC.py` file. The changes primarily focus on handling connection issues more gracefully and ensuring the application can attempt to reconnect if the connection to Discord is lost.

### Improvements to connection handling:

* Added import for `PipeClosed` exception from `pypresence.exceptions` to handle specific pipe closure errors.

* In the `setData` method, added a try-except block to catch `PipeClosed` and `BrokenPipeError` exceptions. If an exception is caught, the method schedules a task to attempt reconnection to Discord after 30 seconds.

* Added a new method `reconnectDiscord` to handle the reconnection logic. This method is called by the scheduled task when a connection error occurs.

* In the `enable` method, added handling for `ConnectionError` exceptions. If a connection error occurs, the method schedules a task to attempt reconnection to Discord after 30 seconds.

HiddenKitten commented:
While this fix is in PR: would it be possible to change at least the error message in enable - specifically, [L232](https://github.com/toontown-archipelago/toontown-archipelago/commit/2cd2cc7461695da060ff2421519bd2d8a5950e97#diff-d6349f13bc81407aa229d9561deb05cd53d5c38cdf28b12382e59ad207f39135R232) to debug instead of warning? if discord happens to not be running, then it'll print a message to log every 5 seconds. (also, it might not be a bad idea to have retries be higher than 5 seconds as well, so we're not constantly attempting so often, in the case discord isn't open)

DarthMDev commented:
Sure!

DarthMDev commented:
This pull request includes several changes to improve the Discord Rich Presence functionality in the `toontown/discord/DiscordRPC.py` file. The changes primarily focus on handling connection issues more gracefully and ensuring the application can attempt to reconnect if the connection to Discord is lost.

### Improvements to connection handling:

* Added import for `PipeClosed` exception from `pypresence.exceptions` to handle specific pipe closure errors.

* In the `setData` method, added a try-except block to catch `PipeClosed` and `BrokenPipeError` exceptions. If an exception is caught, the method schedules a task to attempt reconnection to Discord after 30 seconds.

* Added a new method `reconnectDiscord` to handle the reconnection logic. This method is called by the scheduled task when a connection error occurs.

* In the `enable` method, added handling for `ConnectionError` exceptions. If a connection error occurs, the method schedules a task to attempt reconnection to Discord after 30 seconds.
